src/duckdb.rs

```diff
@@ -346,6 +352,21 @@ impl TableProviderFactory for DuckDBTableProviderFactory {
             .create()
             .map_err(to_datafusion_error)?;
 
+        // If the table is already created, we don't create it again and don't apply primary keys.
+        // We need to verify that the table was created with the correct primary keys.
+        if !duckdb
+            .verify_primary_keys_match()
+            .await
+            .map_err(to_datafusion_error)?
+        {
+            tracing::warn!(
```

Comment:
As a next step we might want to add more control similar to [unsupported_type_action](https://spiceai.org/docs/reference/spicepod/datasets#unsupported_type_action)